src/index.js

```diff
@@ -1,21 +1,40 @@
 const core = require('@actions/core');
-const wait = require('./wait');
+const { App } = require("@octokit/app");
 
-
-// most @actions toolkit packages have async methods
 async function run() {
-  try {
-    const ms = core.getInput('milliseconds');
-    core.info(`Waiting ${ms} milliseconds ...`);
 
-    core.debug((new Date()).toTimeString()); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
-    await wait(parseInt(ms));
-    core.info((new Date()).toTimeString());
+    try {
+        const REPO_OWNER = core.getInput('repo-owner');
+        const REPO_NAME = core.getInput('repo-name');
 
-    core.setOutput('time', new Date().toTimeString());
-  } catch (error) {
-    core.setFailed(error.message);
-  }
+        // Log in
+        const app = new App({
+            appId: process.env.APP_ID,
+            privateKey: process.env.APP_PRIVATE_KEY,
+        });
+        
+        // List all the repository where the App is installed
+        for await (const { octokit, repository } of app.eachRepository.iterator()) {
+            if (repository.owner.login === REPO_OWNER && repository.name === REPO_NAME) {
+                const runners = (await octokit.request('GET /repos/{owner}/{repo}/actions/runners', {
+                    owner: REPO_OWNER,
+                    repo: REPO_NAME
+                })).data.runners;
+                // Delete all the offline self-hosted runners from the repository
+                for (const runner of runners) {
+                    if (runner.status === "offline" ) {
+                        await octokit.request('DELETE /repos/{owner}/{repo}/actions/runners/{runner_id}', {
```

Comment:
Also here, did you try something like [this](https://github.com/pavlovic-ivan/ephemeral-github-runner/blob/main/cleanup-runners.ts#L25)?